project/platform/src/net/ashald/envfile/platform/ui/EnvFileConfigurationEditor.java

```diff
@@ -104,24 +104,37 @@ public static void writeExternal(@NotNull RunConfigurationBase configuration, @N
         }
     }
 
-    public static void patchEnvironmentVariables(@NotNull RunConfigurationBase runConfigurationBase, Map<String, String> environmentVariables) throws ExecutionException {
+    public static void patchEnv(@NotNull RunConfigurationBase runConfigurationBase, Map<String, String> currentEnv) throws ExecutionException {
+        Map<String, String> newEnv = EnvFileConfigurationEditor.collectEnv(runConfigurationBase, currentEnv);
+        currentEnv.clear();
+        currentEnv.putAll(newEnv);
+    }
+
+    public static Map<String, String> collectEnv(@NotNull RunConfigurationBase runConfigurationBase, Map<String, String> userEnv) throws ExecutionException {
```

Comment:
Let's make `collectEnv` our main interface and drop the `patchEnv` defined above - I'll comment below about how we will use it.

Reply:
Done